### PR TITLE
Wrap JenkinsScheduler run body in a try/catch/finally.

### PR DESCRIPTION
**Pull request by Vlatombe**
Split from #179 

If any RuntimeException is raised during Scheduler execution, the object
was left in a corrupt state with running == true and driver != null.

The previous implementation would let any RuntimeException kill the
thread without any trace of it. Now, it is properly logged when it
happens and leave the scheduler in a correct state.

@reviewbybees

### CODE REVIEW

**Comment by carlossg** [comment]
:bee:

File: src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java [hunk 1]
```diff
@@ -114,42 +114,46 @@ public synchronized void init() {
...
+                driver = new MesosSchedulerDriver(JenkinsScheduler.this, framework, mesosCloud.getMaster());
+            }
+            Status runStatus = driver.run();
+            if (runStatus != Status.DRIVER_STOPPED) {
+                LOGGER.severe("The Mesos driver was aborted! Status code: " + runStatus.getNumber());
+            }
+        } catch(RuntimeException e) {
+            LOGGER.log(Level.SEVERE, "Caught a RuntimeException", e);
+        } finally {
+            driver = null;
```
Comment (vinodkone) [inline src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java]: i would also do a `driver.abort()` if `driver != null`.` this ensures this plugin doesn't hoard offers.

**Comment by stephenc** [comment]
:bee:

**Comment by Vlatombe** [comment]
@reviewbybees done

**Pull request merged**